packages/gateway/src/metrics.js

```diff
@@ -0,0 +1,74 @@
+/* global Response caches */
+
+import { METRICS_CACHE_MAX_AGE } from './constants.js'
+import { histogram } from './durable-objects/metrics.js'
+
+/**
+ * Retrieve metrics in prometheus exposition format.
+ * https://prometheus.io/docs/instrumenting/exposition_formats/
+ * @param {Request} request
+ * @param {import('./env').Env} env
+ * @param {import('./index').Ctx} ctx
+ * @returns {Promise<Response>}
+ */
+export async function metricsGet(request, env, ctx) {
+  // TODO: Set cache
+  // const cache = caches.default
+  // let res = await cache.match(request)
+
+  // if (res) {
+  //   return res
+  // }
+  let res
+
+  const id = env.metricsDurable.idFromName('metrics')
+  const stub = env.metricsDurable.get(id)
+
+  const stubResponse = await stub.fetch(request)
+  /** @type {Record<string,import('./durable-objects/metrics').GatewayMetrics>} */
+  const metricsPerGw = await stubResponse.json()
+
+  const metrics = [
+    `# HELP nftstorage_gateway_total_requests Total requests performed.`,
+    `# TYPE nftstorage_gateway_total_requests counter`,
+    ...env.ipfsGateways.map((gw) => [
+      `nftstorage_gateway_total_requests{gateway="${gw}"} ${metricsPerGw[gw].totalReqCount}`,
+    ]),
```

Comment:
An array of array of 1 item?

Reply:
you mean `env.ipfsGateways`? 

It is: [ipfs.io, cf-ipfs.com, pinata] https://ipfs-staging.nft.storage/metrics

Reply:
The map function is returning an array so you'll get an array of arrays.

Reply:
It works out because `toString` on an array of 1 string will just return the string...but it should probably be fixed.